external/scalapack.cmake

```diff
@@ -0,0 +1,83 @@
+
+# -*- mode: cmake -*-
+
+###################
+# Find dependencies related to ScaLAPACK
+###################
+
+if( NOT TARGET scalapackpp::scalapackpp )
+  find_package( scalapackpp CONFIG QUIET HINTS ${SCALAPACKPP_ROOT_DIR} )
```

Comment:
does scalapacpp track the scalapack version? I presume I don't need to worry about this here: as long as the interface is detected it's good to go.

Also, does scalapapp check that the "current" MPI lib and the MPI lib used to compile ScaLAPACK match?

Reply:
> does scalapacpp track the scalapack version?

No, much like the BLAS/LAPACK standards, ScaLAPACK has had a uniform (calling) API since its inception. The version of ScaLAPACK that scalapackpp/blacspp link to is irrelevant (at long as it can resolve the symbols).

> Also, does scalapapp check that the "current" MPI lib and the MPI lib used to compile ScaLAPACK match?

Per our offline discussion no, but unfortunately I think this is going to just have to be left up to the user the make sure that everything is consistent. If ScaLAPACK is compiled static, this shouldn't matter (mod MKL) as the typical netlib impl just makes simple MPI API calls and doesn't depend on anything vendor specific. If MKL is used, my FindBLAS/LAPACK/ScaLAPACK.cmake modules will handle the proper linkage. From blacspp/scalapackpp's perspective, it doesn't matter as long as it can fully resolve the necessary symbols at link.

So the only issue I can think of would be a shared compilation of netlib ScaLAPACK s.t. that it points to a specific MPI impl. We could in principle just insist that ScaLAPACK is compiled static(?)